pkg/e2e/setup.go

```diff
@@ -71,11 +76,88 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 	// only needs to run once
 })
 
-// Collect logs after each test
-var _ = ginkgo.AfterSuite(func() {
+// After all tests run, clean up within the Ginkgo context
+var _ = ginkgo.SynchronizedAfterSuite(func() {
+	h := helper.NewOutsideGinkgo()
+	// Run Must Gather
+	func() {
+		defer ginkgo.GinkgoRecover()
+
+		log.Print("Running Must Gather...")
+		mustGatherTimeoutInSeconds := 900
+		h.SetServiceAccount("system:serviceaccount:%s:cluster-admin")
+		r := h.Runner(fmt.Sprintf("oc adm must-gather --dest-dir=%v", runner.DefaultRunner.OutputDir))
+		r.Name = "must-gather"
+		r.Tarball = true
+		stopCh := make(chan struct{})
+		err := r.Run(mustGatherTimeoutInSeconds, stopCh)
+		Expect(err).NotTo(HaveOccurred())
+		gatherResults, err := r.RetrieveResults()
+		Expect(err).NotTo(HaveOccurred())
+		h.WriteResults(gatherResults)
+
+	}()
+
+	// Run Cluster State
+	func() {
+		defer ginkgo.GinkgoRecover()
+
+		log.Print("Gathering Cluster State...")
+		clusterState := h.GetClusterState()
+		stateResults := make(map[string][]byte, len(clusterState))
+		for resource, list := range clusterState {
+			data, err := json.MarshalIndent(list, "", "    ")
+			Expect(err).NotTo(HaveOccurred())
+
+			var gbuf bytes.Buffer
+			zw := gzip.NewWriter(&gbuf)
+			_, err = zw.Write(data)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = zw.Close()
+			Expect(err).NotTo(HaveOccurred())
+
+			// include gzip in filename to mark compressed data
+			filename := fmt.Sprintf("%s-%s-%s.json.gzip", resource.Group, resource.Version, resource.Resource)
+			stateResults[filename] = gbuf.Bytes()
+		}
+
+		// write results to disk
+		h.WriteResults(stateResults)
+	}()
+
+	// Get state from OCM
+	func() {
```

Comment:
I don't know that this is necessary since we're not using ginkgo.GinkgoRecover() in here.